hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSDirRenameOp.java

```diff
@@ -400,6 +408,13 @@ static RenameResult unprotectedRenameTo(FSDirectory fsd,
       throw new ParentNotDirectoryException(error);
     }
 
+    if (snapshottableDirs.size() > 0) {
+      if (dstParent.asDirectory().isDescendantOfSnapshotRoot()) {
+        throw new SnapshotException("Failed to rename " + src + " to " + dst +
+                " due to nested snapshottable directory");
+      }
+    }
+
```

Comment:
This code snippet is written twice in same class, in different methods. Can we instead carve it into a method to avoid code duplication?